### PR TITLE
Use config when starting development server

### PR DESCRIPTION
**Pull request by librarianmage**
[//]: # (Note: See http://habitica.wikia.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API for more info)

[//]: # (Put Issue # or URL here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes N/A

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)

This PR allows the webpack server to use the user's config.json in determining the proxy locations, allowing the user to change the port and the ip of the Express.js server.

[//]: # (Put User ID in here - found on the Habitica website at User Icon > Settings > API)

----
UUID: 95f16006-1504-4a9f-95b6-0c7314dbd7ee

### CODE REVIEW

File: webpack/config/index.js [hunk 1]
```diff
@@ -3,6 +3,17 @@ const path = require('path');
 const staticAssetsDirectory = './website/static/.'; // The folder where static files (not processed) live
 const prodEnv = require('./prod.env');
 const devEnv = require('./dev.env');
+const nconf = require('nconf');
+
+const PATH_TO_CONFIG = path.join(path.resolve(__dirname, '../../config.json'));
+let configFile = PATH_TO_CONFIG;
+
+nconf
```
Comment (paglias) [inline webpack/config/index.js]: this looks good but we have almost the same code in website/server/libs/setupNconf, can we reuse it?

**Comment by librarianmage** [comment]
@paglias I don't think so because Webpack gets tripped up by the import
statements.

**Comment by paglias** [comment]
What do you mean by tripped? You should use require non import

Il 06 Nov 2017 15:42, "MathWhiz" <notifications@github.com> ha scritto:

**Comment by librarianmage** [comment]
the import statements in the file tripped up webpack

```
(function (exports, require, module, __filename, __dirname) { import nconf from 'nconf';
                                                              ^^^^^^
SyntaxError: Unexpected token import
    at createScript (vm.js:56:10)
    at Object.runInThisContext (vm.js:97:10)
    at Module._compile (module.js:542:28)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/home/MathWhiz/habitrpg/webpack/config/prod.env.js:2:20)
```

https://github.com/HabitRPG/habitica/blob/12fd79059b09f258bf9f7d54415180f03a2c8ba8/website/server/libs/setupNconf.js#L1-L2

**Comment by paglias** [comment]
I see, in that case I think we should change the setupNconf file to use `require` with a comment about why it's necessary so it can be used by webpack too

**Comment by librarianmage** [comment]
Will it work there?

**Comment by paglias** [comment]
yep, require works everywhere, import only in /server because it needs babel for transpilation

**Comment by librarianmage** [comment]
@paglias Like this?

**Comment by paglias** [comment]
yep, this looks good! there are some linting issues though

**Comment by librarianmage** [comment]
Updated!

**Review by paglias — APPROVED** [review]

**Pull request merged**